### PR TITLE
fix(inputs.kube_inventory): send file location to enable token auto-refresh

### PR DESCRIPTION
**Pull request by powersj**
Currently, Telegraf will read a token file and provide that as a string
to the K8s client. This will prevent the K8s client from attempting to
auto-refresh the token. In newer versions of K8s, tokens will expire
every hour as a security measure.

This, instead will pass the file location rather than having telegraf
read it directly. Because of this new K8s behavior, this also deprecates
and warns the user when using a hard-coded string token.

fixes: #11267

### CODE REVIEW

File: plugins/inputs/kube_inventory/sample.conf [hunk 1]
```diff
@@ -7,9 +7,13 @@
...
 
   ## Use bearer token for authorization. ('bearer_token' takes priority)
+  ##
   ## If both of these are empty, we'll use the default serviceaccount:
   ## at: /run/secrets/kubernetes.io/serviceaccount/token
-  # bearer_token = "/path/to/bearer/token"
+  ##
+  ## To auto-refresh the token, please use a file with the bearer_token option.
+  ## If given a string, Telegraf cannot refresh the token periodically.
+  # bearer_token = "/run/secrets/kubernetes.io/serviceaccount/token"
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/sample.conf:16]: Can you please also add a deprecation warning here!?

**Review by srebhan — COMMENTED** [review]
Looks good to me. I only think it's helpful to add a sentence about deprecation of `bearer_token` to the sample.conf file.

File: plugins/inputs/kube_inventory/sample.conf (hunk 1 above)
Comment (powersj) [reply plugins/inputs/kube_inventory/sample.conf:16]: woops, yep pushed an update for `bearer_token_string`

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @powersj!

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**